File: scripts/workflow-paths.ts
import { existsSync } from "node:fs";
import path from "node:path";

export const siteRoot = path.resolve(process.cwd());

function resolveWorkflowRoot() {
  const configured = process.env.KK_WORKFLOW_ROOT?.trim();
  if (configured) {
    return path.resolve(configured);
  }

  const siblingRoot = path.resolve(siteRoot, "..", "KK Diction");
  return existsSync(siblingRoot) ? siblingRoot : siteRoot;
}

export const workflowRoot = resolveWorkflowRoot();
export const dbImageRoot = path.join(workflowRoot, "db image");
export const cutRoot = path.join(dbImageRoot, "cut");
export const dbModsRoot = path.join(workflowRoot, "db mods");

export function resolveWorkflowPath(relativePath: string) {
  const resolved = path.resolve(workflowRoot, relativePath);
  const relative = path.relative(workflowRoot, resolved);
  if (relative.startsWith("..") || path.isAbsolute(relative)) {
    throw new Error(`Path must stay inside KK_WORKFLOW_ROOT: ${relativePath}`);
  }
  return resolved;
}

File: scripts/workflow_paths.py
from __future__ import annotations

import os
from pathlib import Path


SITE_ROOT = Path.cwd().resolve()


def resolve_workflow_root() -> Path:
    configured = os.environ.get("KK_WORKFLOW_ROOT", "").strip()
    if configured:
        return Path(configured).resolve()

    sibling_root = (SITE_ROOT.parent / "KK Diction").resolve()
    return sibling_root if sibling_root.exists() else SITE_ROOT


WORKFLOW_ROOT = resolve_workflow_root()
DB_IMAGE_ROOT = WORKFLOW_ROOT / "db image"
CUT_ROOT = DB_IMAGE_ROOT / "cut"
DB_MODS_ROOT = WORKFLOW_ROOT / "db mods"
